src/python/pants/core/util_rules/archive.py

```diff
@@ -187,13 +39,15 @@ class CreateArchive:
 @rule(desc="Creating an archive file", level=LogLevel.DEBUG)
 async def create_archive(request: CreateArchive) -> Digest:
     if request.format == ArchiveFormat.ZIP:
-        zip_binary = await Get(ZipBinary, _ZipBinaryRequest())
-        argv = zip_binary.create_archive_argv(request)
+        zip_binary = await Get(ZipBinary, ZipBinaryRequest())
```

Comment:
`zip_binary` and `tar_binary` should probably be injected (possibly involving a refactor to make that more efficient). This is possibly better as follow-up work than something for this PR.

Reply:
The laziness is intentional: https://github.com/pantsbuild/pants/pull/12947